src/warc2zim/content_rewriting/html.py

```diff
@@ -161,3 +139,50 @@ def handle_pi(self, data: str):
 
     def unknown_decl(self, data: str):
         self.handle_decl(data)
+
+    def process_attr(
+        self,
+        attr: tuple[str, str | None],
+        url_rewriter: UrlRewriterProto,
+    ) -> tuple[str, str | None]:
+        if not attr[1]:
+            return attr
+
+        if attr[0] in ("href", "src"):
+            if self.html_rewrite_context == "js-module":
+                self.notify_js_module(self.url_rewriter.get_item_path(attr[1]))
+            return (attr[0], url_rewriter(attr[1]))
+        if attr[0] == "srcset":
+            value_list = attr[1].split(",")
+            new_value_list = []
+            for value in value_list:
+                url, *other = value.strip().split(" ", maxsplit=1)
+                new_url = url_rewriter(url)
+                new_value = " ".join([new_url, *other])
+                new_value_list.append(new_value)
+            return (attr[0], ", ".join(new_value_list))
+        if attr[0] == "style":
+            return (attr[0], self.css_rewriter.rewrite_inline(attr[1]))
+        return attr
+
+    def format_attr(self, name: str, value: str | None) -> str:
+        if value is None:
+            return name
+        html_escaped_value = escape(value, quote=True)
+        return f'{name}="{html_escaped_value}"'
+
+    def transform_attrs(
+        self,
+        attrs: AttrsList,
+        url_rewriter: UrlRewriterProto,
+    ) -> str:
+        processed_attrs = (self.process_attr(attr, url_rewriter) for attr in attrs)
+        return " ".join(self.format_attr(*attr) for attr in processed_attrs)
+
+    def extract_attr(
+        self, attrs: AttrsList, name: str, default: str | None = None
+    ) -> str | None:
+        for attr_name, value in attrs:
+            if attr_name == name:
+                return value
+        return default
```

Comment:
You don't help us to review this change by moving to function code elsewhere and updating it in the same time. Moving the free function as a method and then updating it would have been simpler.

Reply:
Agreed, sorry for that. Do you want that I make this move in another PR first? (but then I will have to force-push in this PR branch to do the rebase)